### PR TITLE
Feature/target av duration

### PR DESCRIPTION
**Pull request by edsilv**
Add a second param to set target for MediaElement in order to pause after a given duration.

### CODE REVIEW

File: src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts [hunk 1]
```diff
@@ -59,12 +59,32 @@ export class MediaElementCenterPanel extends CenterPanel<
...
+          }
+
+          this.player.setCurrentTime(startTime);
+          this.player.play();
+
+          const duration = (endTime - startTime) * 1000;
+
+          setTimeout(() => {
+            this.player.pause();
+          }, duration);
```
Comment (stephenwf) [inline src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts:81]: Maybe need to cancel this timeout if `SET_TARGET` is called again, or if it's manually paused.

Comment (demiankatz) [reply src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts:81]: Good point. And what should the behavior be if you unpause after pausing? Do we still expect the range to be respected, or does interrupting it cause the full media to play through?

Comment (edsilv) [reply src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts]: Have added some new logic to clear the timeout on new SET_TARGET or pause. Not sure what to do if unpaused again... I can live with it playing through to the end.
I've also added an autoPlayOnSetTarget config option which is necessary for my use case.

File: src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts [hunk 2]
```diff
@@ -59,12 +60,42 @@ export class MediaElementCenterPanel extends CenterPanel<
...
+            }, duration);
+          }
+
+          return;
+        }
       }
-      that.player.setCurrentTime(t);
-      that.player.play();
+      this.player.setCurrentTime(t);
+      this.player.play();
```
Comment (demiankatz) [inline src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts:98]: Do you also need to account for `this.config.options.autoPlayOnSetTarget` here, or is there a reason this case is different?

Comment (edsilv) [reply src/content-handlers/iiif/modules/uv-mediaelementcenterpanel-module/MediaElementCenterPanel.ts]: have pushed a fix

**Review by demiankatz — APPROVED** [review]
Thanks, @edsilv. I'm not in a position to actually test this hands-on, but it looks good to me in theory now at least. :-)

**Comment by LanieOkorodudu** [comment]
@edsilv @demiankatz I attempted to test this configuration on the Vercel app using the available Video and Audio elements. Based on my understanding, this setup should allow users to specify a playback duration (e.g., 20 seconds), after which the media should automatically pause. Could you provide a sample manifest or configuration that I can use to validate if this functionality is working as expected? Please let me know if my interpretation of the feature is incorrect.

**Comment by demiankatz** [comment]
@LanieOkorodudu, I'm not really familiar with how the target functionality works in AV; I assume you can test this by putting something in the target box in the example page, but I'm not sure what syntax to use. Perhaps @stephenwf can suggest something if @edsilv is unavailable.

**Comment by edsilv** [comment]
@LanieOkorodudu Here's the preview with an AV manifest loaded: https://universalviewer-git-feature-target-av-duration-bl-uv-team.vercel.app/#?xywh=&iiifManifestId=https%3A%2F%2Fbiiif-template-example-3kntb3jpl-mnemoscene.vercel.app%2Fvideo%2Findex.json

In the target field under the UV, try setting: https://biiif-template-example-3kntb3jpl-mnemoscene.vercel.app/video/index.json/canvas/0#t=2

Which should play through to the end starting from 2 seconds.

Then try setting:

https://biiif-template-example-3kntb3jpl-mnemoscene.vercel.app/video/index.json/canvas/0#t=2,4

Which should start playing at 2 seconds then stop at 4 seconds.

**Comment by LanieOkorodudu** [comment]
> @LanieOkorodudu Here's the preview with an AV manifest loaded: https://universalviewer-git-feature-target-av-duration-bl-uv-team.vercel.app/#?xywh=&iiifManifestId=https%3A%2F%2Fbiiif-template-example-3kntb3jpl-mnemoscene.vercel.app%2Fvideo%2Findex.json
> 
> In the target field under the UV, try setting: https://biiif-template-example-3kntb3jpl-mnemoscene.vercel.app/video/index.json/canvas/0#t=2
> 
> Which should play through to the end starting from 2 seconds.
> 
> Then try setting:
> 
> https://biiif-template-example-3kntb3jpl-mnemoscene.vercel.app/video/index.json/canvas/0#t=2,4
> 
> Which should start playing at 2 seconds then stop at 4 seconds.

Thank you, @edsilv . I have tested the implementation, initially setting the start point at 2 seconds as suggested, and also evaluated the second proposed approach. Additionally, I verified the functionality with an audio file to ensure compatibility. I tested it using the following link: [https://biiif-template-example-3kntb3jpl-mnemoscene.vercel.app/audio/index.json](One Small Step for Man - Audio), and confirmed that it works as expected.

Thanks again, Eds. I’m confident that users will find this feature valuable.

**Comment by LanieOkorodudu** [comment]
> @LanieOkorodudu, I'm not really familiar with how the target functionality works in AV; I assume you can test this by putting something in the target box in the example page, but I'm not sure what syntax to use. Perhaps @stephenwf can suggest something if @edsilv is unavailable.

@demiankatz Thanks Demian, I have tested the feature in video as suggested by Eds and additionally verified it with audio to ensure functionality across both formats. I can confirm that it works as expected.

**Comment by demiankatz** [comment]
Thanks, everyone, I'll go ahead and merge this!

**Pull request merged**